### PR TITLE
Modal to force username verification

### LINKED ISSUE #10697: WEB [Nov 2] Implement modal that forces users to change username
Once we release the username feature, users have to set/confirm their username

Fixes needed from review:

- [x] Allow modal to grow correctly to accommodate validation errors
- [x] Check on why z-indexing issue sometimes recurs
- [x] Implement modal stack quick fix suggested by Matteo
- [x] Move email validations to mixin using `validator` library https://github.com/HabitRPG/habitica/pull/10757#commitcomment-31038859
- [x] Remove unnecessary CSS prefixes https://github.com/HabitRPG/habitica/pull/10757#commitcomment-31038852
- [x] Allow length-1 usernames https://github.com/HabitRPG/habitica/pull/10757#pullrequestreview-166752463
- [x] Try another modal stack fix, sigh
- [x] Only import relevant methods from `validator`

### PR DESCRIPTION
**Pull request by SabreCat**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

**NOT FOR PRODUCTION AT THIS TIME** - This is meant to go live when we are ready to require usernames for all users, in preparation for full feature rollout.

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)

Fixes #10697 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

* Adds a modal that appears automatically if the user has not yet verified their username. The modal gives the user the opportunity to change their display name and/or username and cannot be dismissed except by providing valid input for both fields.
* Adds an API route `verify-display-name`, based on `verify-username`, so we can check display names for slurs and length inline.
* Cleans out a few unused name-related strings.

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

### CODE REVIEW

**Comment by SabreCat** [comment]
If the "verify display name" implementation looks OK, I can implement it over on the Settings page as well.

File: website/client/components/settings/verifyUsername.vue [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+  import debounce from 'lodash/debounce';
+  import helloNametag from 'assets/svg/hello-habitican.svg';
+  import { mapState } from 'client/libs/store';
+
+  export default {
+    computed: {
+      ...mapState({
+        user: 'user.data',
+      }),
+      displayNameInvalid () {
```
Comment (paglias) [inline website/client/components/settings/verifyUsername.vue:164]: I think usernames and display names with a lenght of 1 should be valid, here they're valid only with at least 2 chars

**Comment by paglias** [comment]
I'm having an issue where on this branch the RYA modal never opens

**Comment by SabreCat** [comment]
I'm going to combine this PR with the other one, as they're all going live at the same time and I otherwise have to keep merging them together to test, hah

**Pull request closed without merging**